civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -4675,6 +4676,52 @@ function civihr_employee_portal_form_user_profile_form_alter(&$form, &$form_stat
 }
 
 /**
+ * Implements hook_file_presave().
+ *
+ * @param \stdClass $file
+ */
+function civihr_employee_portal_file_presave($file) {
+  $resizeSources = [
+    'submitted_civicrm_1_contact_1_contact_image_URL'
+  ];
+
+  if (in_array($file->source, $resizeSources)) {
+    ImageHelper::resizeForProfile($file->uri);
+  }
+}
+
+/**
+ * @param string $formName
+ * @param CRM_Contact_Form_Contact $form
+ */
+function civihr_employee_portal_civicrm_postProcess($formName, &$form) {
+  if ($formName !== 'CRM_Contact_Form_Contact') {
+    return;
+  }
+
+  $contactID = $form->getContactID();
+  $imageChanged = !empty($form->_submitFiles['image_URL']['name']);
+
+  if (!$imageChanged) {
+    return;
+  }
+
+  $targetContact = civicrm_api3('Contact', 'getsingle', ['id' => $contactID]);
+  // Since this is postProcess image_URL is already pointing to new image
+  $imageUrl = $targetContact['image_URL'];
+
+  // Get the full path to the image
+  $queryString = parse_url($imageUrl, PHP_URL_QUERY);
```

Comment:
the lines from 4714 to 4719 can go into separate function **_civihr_employee_portal_get_image_path_from_contact_image_url($imageUrl)**

Reply:
Moved them to ImageHelper class, I already know another ticket where I'll be using this function.